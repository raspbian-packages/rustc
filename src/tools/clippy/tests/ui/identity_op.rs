#![feature(plugin)]
#![plugin(clippy)]

const ONE : i64 = 1;
const NEG_ONE : i64 = -1;
const ZERO : i64 = 0;

#[allow(eq_op, no_effect, unnecessary_operation, double_parens)]
#[warn(identity_op)]
fn main() {
    let x = 0;

    x + 0;
    x + (1 - 1);
    x + 1;
    0 + x;
    1 + x;
    x - ZERO;     //no error, as we skip lookups (for now)
    x | (0);
    ((ZERO)) | x; //no error, as we skip lookups (for now)

    x * 1;
    1 * x;
    x / ONE;      //no error, as we skip lookups (for now)

    x / 2;        //no false positive

    x & NEG_ONE;  //no error, as we skip lookups (for now)
    -1 & x;
}
