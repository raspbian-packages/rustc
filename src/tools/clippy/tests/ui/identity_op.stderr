error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:13:5
   |
13 |     x + 0;
   |     ^^^^^
   |
   = note: `-D identity-op` implied by `-D warnings`

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:14:5
   |
14 |     x + (1 - 1);
   |     ^^^^^^^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:16:5
   |
16 |     0 + x;
   |     ^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:19:5
   |
19 |     x | (0);
   |     ^^^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:22:5
   |
22 |     x * 1;
   |     ^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:23:5
   |
23 |     1 * x;
   |     ^^^^^

error: the operation is ineffective. Consider reducing it to `x`
  --> $DIR/identity_op.rs:29:5
   |
29 |     -1 & x;
   |     ^^^^^^

error: the operation is ineffective. Consider reducing it to `u`
  --> $DIR/identity_op.rs:32:5
   |
32 |     u & 255;
   |     ^^^^^^^

error: aborting due to 8 previous errors

